### PR TITLE
chore: bump crate versions and update changelogs for #5676

### PR DESCRIPTION
**Pull request by hanabi1224**
## Description

This PR bumps crate versions and add changelog entries for crates that are changed in #5676

Question: When should a crate version bump in the current release process? Should it be right before or right after publishing? I see most of current crate versions are published while some are not (e.g. libp2p-autonat@0.13.1 libp2p-gossisub@0.48.0 and libp2p-perf@0.4.0 etc.)

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Comment by hanabi1224** [comment]
Hmm, when I update `libp2p-identity` version in root `Cargo.toml`, I got [semver CI failure](https://github.com/libp2p/rust-libp2p/actions/runs/11852347264/job/33030360152), when not, I got [test identity CI failure](https://github.com/libp2p/rust-libp2p/actions/runs/11854436745/job/33036614219?pr=5678), what should I do to bump the version of this crate?

File: Cargo.toml [hunk 1]
```diff
@@ -75,33 +75,33 @@ rust-version = "1.75.0"
...
-libp2p-core = { version = "0.42.0", path = "core" }
-libp2p-dcutr = { version = "0.12.0", path = "protocols/dcutr" }
+libp2p-connection-limits = { version = "0.4.1", path = "misc/connection-limits" }
+libp2p-core = { version = "0.42.1", path = "core" }
+libp2p-dcutr = { version = "0.12.1", path = "protocols/dcutr" }
 libp2p-dns = { version = "0.42.0", path = "transports/dns" }
 libp2p-floodsub = { version = "0.45.0", path = "protocols/floodsub" }
 libp2p-gossipsub = { version = "0.48.0", path = "protocols/gossipsub" }
 libp2p-identify = { version = "0.46.0", path = "protocols/identify" }
 libp2p-identity = { version = "0.2.9" }
```
Comment (jxs) [inline Cargo.toml]:
```suggestion
libp2p-identity = { version = "0.2.10" }
```

**Review by jxs — APPROVED** [review]
> Hmm, when I update `libp2p-identity` version in root `Cargo.toml`, I got [semver CI failure](https://github.com/libp2p/rust-libp2p/actions/runs/11852347264/job/33030360152), when not, I got [test identity CI failure](https://github.com/libp2p/rust-libp2p/actions/runs/11854436745/job/33036614219?pr=5678), what should I do to bump the version of this crate?

Hi, yeah it's expected with libp2p-identity due to the way it is imported. See [here](https://github.com/libp2p/rust-libp2p/actions/runs/11935324073/job/33266403454?pr=5668) for a CI that has run after `libp2p-identity` `0.2.10` was published.
Thanks!

**Comment by jxs** [comment]
published `libp2p-identity` version `0.2.10`

**Pull request merged**